pkg/services/sqlstore/datasource.go

```diff
@@ -27,6 +27,23 @@ func init() {
 	bus.AddHandler("sql", GetDataSourceByName)
 }
 
+func (ss *SqlStore) GetDataSourceByID(id, orgID int64) (*models.DataSource, error) {
```

Comment:
Shouldn't we also make it so that the query-based version of this method uses this method? Seems like a bad idea to have two separate implementations. From what I can tell, the difference is that one of them uses the global `x.Get` which I don't think is something we really want to do anyway.

Reply:
@xlson Let me look into this.

Reply:
@xlson Solved it by making a separate common function.